contents/docs/auth.mdx

```diff
@@ -127,37 +125,49 @@ export function handleMutate(request: Request) {
   )
 
   if (!session) {
+    // can be 401 or 403
     return json({error: 'Unauthorized'}, {status: 401})
   }
 
   // handle mutate request ...
 }
 ```
 
-## Auth Failure and Refresh
+To mark a request as unauthorized, return a `401` or `403` status code from your [queries](/docs/queries) or [mutate](/docs/mutators) endpoint.
 
-To mark a request as unauthorized, return a `401` or `403` status code from your [queries](/docs/queries) or [mutate](/docs/mutators) endpoint:
+This will cause Zero to disconnect from `zero-cache` and the [connection status](/docs/TODO) will change to `needs-auth`. You can then re-authenticate the user and call `zero.connection.connect()` to reconnect to `zero-cache`:
 
-```ts
-export function handleMutate(request: Request) {
-  const session = await authenticate(request)
+```tsx
+function NeedsAuthDialog() {
+  const zero = useZero()
+  const connectionState = useZeroConnectionState();
 
-  if (!session) {
-    return json({error: 'Forbidden'}, {status: 403})
+  const refreshCookie = async () => {
+    const cookie = await login();
```

Comment:
Typically in such systems the cookie would not be returned to client for security reasons (it would be http-only). There would be some other value returned though.

I think for this snippet, can just say `await login()` with no ret val.